src/selement/selementAssociation.ts

```diff
@@ -6,6 +6,42 @@ class SElementAssociation {
   deptype?: string;
   attrs?: any;
 
+  /**
+     Create association between two elements if there already does not exist a similar association.
+     The association is considered to be similar if toElement has an incoming association with the
+     same type and the same fromElement.
+     @param {SElement} fromElement the elemenet that is the starting point of the association
+     @param {SElement} toElement the element that is the ending point of the association
+     @param {string} deptype the type of the association
+     @param {any} depattrs attributes for the associtaion
+     @returns {boolean} Return true if new association was created, false if new association was not
+     created due to an existing similar association
+   */
+  static createUniqueElementAssociation(
+    fromElement: SElement,
+    toElement: SElement,
+    deptype?: any,
+    depattrs: any = {}
+  ): boolean {
+    const existingAssociations = toElement.incoming.filter((incoming) => {
+      const fromElementMatches = incoming.fromElement === fromElement;
+      return deptype
+        ? deptype === incoming.deptype && fromElementMatches
+        : fromElementMatches;
+    });
+    // Do not create association if the same association already exists
+    if (existingAssociations.length > 0) {
+      return false;
```

Comment:
Is it possible to return first of the matching existingAssocations element and false here?